File: library/src/com/google/maps/android/kml/Point.java
package com.google.maps.android.kml;

import com.google.android.gms.maps.model.LatLng;

/**
 * Created by lavenderch on 12/22/14.
 */
public class Point implements Geometry {

    private LatLng pointCoordinate;

    /**
     * Receives text representing a tuple coordinates separated by a comma
     * (longitude, latitude, altitude) This method converts this tuples into LatLng points,
     * and ignores the altitude component
     * @param text A string representation of a tuple coordinate, with values separated by commas
     */
    @Override
    public void createCoordinates(String text) {
        String[] coordinate = text.split(",");
        LatLng pointCoordinate = Placemark.convertToLatLng(coordinate);
        setGeometry(pointCoordinate);
    }

    /**
     * Creates a LatLng point
     *
     * @param geometry  An object which represents a LatLng point
     */
    public void setGeometry(Object geometry) {
        pointCoordinate = ((LatLng) geometry);
    }

    /**
     * Returns a LatLng point representing point of a marker
     */
    public Object getGeometry() {
       return pointCoordinate;
    }
}

File: android-maps-utils-ktx/src/main/java/com/google/maps/ktx/geometry/Point.kt
package com.google.maps.ktx.geometry

import com.google.maps.android.geometry.Point

/**
 * Returns the x value of this Point.
 *
 * e.g.
 *
 * ```
 * val (x, _) = point
 * ```
 */
inline operator fun Point.component1() = this.x

/**
 * Returns the y value of this Point.
 *
 * e.g.
 *
 * ```
 * val (_, y) = point
 */
inline operator fun Point.component2() = this.y
